### PR TITLE
show error when trying to scan non existing path

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@icewind1991 I added the PHPDoc and handled the exception in the scan:appdata command as well.

File: lib/private/Files/Utils/Scanner.php [hunk 1]
```diff
@@ -216,6 +218,9 @@ public function scan($dir = '') {
 			try {
 				$propagator = $storage->getPropagator();
 				$propagator->beginBatch();
+				if (!$storage->file_exists($relativePath)) {
```
Comment (nickvergessen) [inline lib/private/Files/Utils/Scanner.php]: Do this before you begin the transaction 5 lines above?

**Review by nickvergessen — DISMISSED** [review]

File: lib/private/Files/Utils/Scanner.php (hunk 1 above)
Comment (icewind1991) [inline lib/private/Files/Utils/Scanner.php]: done

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by MorrisJobke** [comment]
To test just do something like this: `occ files:scan --path=admin/files2`

**Review by rullzer — APPROVED** [review]
Lets do this!

**Pull request merged**